DNN Platform/Library/Entities/Users/UserEventHandlers.cs

```diff
@@ -51,6 +51,7 @@ public void UserDeleted(object sender, UserEventArgs args)
 
         public void UserApproved(object sender, UserEventArgs args)
         {
+            if (!args.SendNotification) return;
             Mail.SendMail(args.User, MessageType.UserRegistrationPublic, PortalSettings.Current);
             DeleteAllNotifications(args.User.UserID);
```

Comment:
Right, that's my question.  It doesn't immediately seem to me like it should be skipped (but it's definitely possible that it's correct to skip it).